tests/components/forked_daapd/test_media_player.py

```diff
@@ -0,0 +1,755 @@
+        "uri": "library:track:1",
+    }
+]
+
+SAMPLE_PLAYLISTS = [{"id": 7, "name": "test_playlist", "uri": "library:playlist:2"}]
+
+
+@pytest.fixture(name="config_entry")
+def config_entry_fixture():
+    """Create hass config_entry fixture."""
+    data = {
+        CONF_HOST: "192.168.1.1",
+        CONF_PORT: "2345",
+        CONF_PASSWORD: "",
+    }
+    return MockConfigEntry(
+        version=1,
+        domain=DOMAIN,
+        title="",
+        data=data,
+        options={CONF_TTS_PAUSE_TIME: 0},
+        system_options={},
+        source=SOURCE_USER,
+        connection_class=CONN_CLASS_LOCAL_PUSH,
+        entry_id=1,
+    )
+
+
+@pytest.fixture(name="get_request_return_values")
+async def get_request_return_values_fixture():
+    """Get request return values we can change later."""
+    return {
+        "config": SAMPLE_CONFIG,
+        "outputs": SAMPLE_OUTPUTS_ON,
+        "player": SAMPLE_PLAYER_PAUSED,
+        "queue": SAMPLE_QUEUE_TTS,
+    }
+
+
+@pytest.fixture(name="mock_api_object")
+async def mock_api_object_fixture(hass, config_entry, get_request_return_values):
+    """Create mock api fixture."""
+
+    async def get_request_side_effect(update_type):
+        if update_type == "outputs":
+            return {"outputs": get_request_return_values["outputs"]}
+        return get_request_return_values[update_type]
+
+    with patch(
+        "homeassistant.components.forked_daapd.media_player.ForkedDaapdAPI",
+        autospec=True,
+    ) as mock_api:
+        mock_api.return_value.get_request.side_effect = get_request_side_effect
+        mock_api.return_value.full_url.return_value = ""
+        mock_api.return_value.get_pipes.return_value = SAMPLE_PIPES
+        mock_api.return_value.get_playlists.return_value = SAMPLE_PLAYLISTS
+        config_entry.add_to_hass(hass)
+        await config_entry.async_setup(hass)
+        await hass.async_block_till_done()
+
+    mock_api.return_value.start_websocket_handler.assert_called_once()
+    mock_api.return_value.get_request.assert_called_once()
+    updater_update = mock_api.return_value.start_websocket_handler.call_args[0][2]
+    await updater_update(["player", "outputs", "queue"])
+    await hass.async_block_till_done()
+
+    async def add_to_queue_side_effect(
+        uris, playback=None, playback_from_position=None, clear=None
+    ):
+        await updater_update(["queue", "player"])
+
+    mock_api.return_value.add_to_queue.side_effect = (
+        add_to_queue_side_effect  # for play_media testing
+    )
+
+    async def pause_side_effect():
+        await updater_update(["player"])
+
+    mock_api.return_value.pause_playback.side_effect = pause_side_effect
+
+    return mock_api.return_value
+
+
+async def test_unload_config_entry(hass, config_entry, mock_api_object):
+    """Test the player is removed when the config entry is unloaded."""
+    assert hass.states.get(TEST_MASTER_ENTITY_NAME)
+    assert hass.states.get(TEST_ZONE_ENTITY_NAMES[0])
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5518609)